tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -604,7 +604,9 @@ private static void notifyObservers(Span span) {
 
         @Override
         public void tag(Span.Builder target, String key, String value) {
-            target.putMetadata(key, value);
+            if (key != null && value != null) {
+                target.putMetadata(key, value);
+            }
         }
 
         @Override
```

Comment:
Nulls in maps anyone? :)

Reply:
The map case is a little bit special, I don't think we need to handle nulls in that case. The null key/value piece above isn't strictly required, but I'd rather not allow tracing uses to cause 5xx failures in production.

Reply:
Well, the map case delegates to putAll in a conjure generate thing, which disallows nulls, so that'll throw.